File: argus/claude/templates/python/src/qa/pages/login_page.py
"""Example Page Object — adjust locators to the real SPA at engagement start.

Locators use role/label (resilient to styling churn, mirror what a user sees);
methods express intent. ADAPT-ME everywhere marked.
"""
from __future__ import annotations

import re

from playwright.sync_api import Page

from qa.config import ENV
from qa.pages.base_page import BasePage


class LoginPage(BasePage):
    path = "/"  # <-- adapt (e.g. "/login")

    def __init__(self, page: Page) -> None:
        super().__init__(page)
        self.username_input = page.get_by_label(re.compile(r"email|username", re.I))
        self.password_input = page.get_by_label(re.compile(r"password", re.I))
        self.submit_button = page.get_by_role(
            "button", name=re.compile(r"log\s*in|sign\s*in", re.I)
        )

    def login_as(self, role: str) -> None:
        account = ENV.accounts[role]
        self.goto()
        self.username_input.fill(account.username)
        self.password_input.fill(account.password)
        self.submit_button.click()

File: argus/claude/templates/typescript/src/pages/login.page.ts
import { Locator, Page } from '@playwright/test';
import { BasePage } from './base.page';
import { ENV, Role } from '../config/env';

// ADAPT-ME: example Page Object. Adjust locators to the real SPA at engagement start.
export class LoginPage extends BasePage {
  readonly path = '/'; // <-- adapt (e.g. '/login')

  readonly usernameInput: Locator;
  readonly passwordInput: Locator;
  readonly submitButton: Locator;

  constructor(page: Page) {
    super(page);
    this.usernameInput = page.getByLabel(/email|username/i);
    this.passwordInput = page.getByLabel(/password/i);
    this.submitButton = page.getByRole('button', { name: /log\s*in|sign\s*in/i });
  }

  async loginAs(role: Role): Promise<void> {
    const { username, password } = ENV.accounts[role];
    await this.goto();
    await this.usernameInput.fill(username);
    await this.passwordInput.fill(password);
    await this.submitButton.click();
  }
}
